src/main/java/playercreation/PlayerCreatorInputValidator.java

```diff
@@ -72,4 +73,43 @@ else if (input.equalsIgnoreCase("return")) {
         }
         return null;
     }
+
+    /**
+     * Returns a specific error message based on the input and the current question.
+     * @param input The invalid user input.
+     * @return A String of a specific error message based on the invalid input.
+     */
+    @Override
+    public String getErrorMessage(String input) {
+        if (currQuestion == PlayerQuestion.NAME) {
```

Comment:
looks like you have multiple blocks doing a choice based on an enum. Maybe split these into separate classes?